api/grpcserver/config.go

```diff
@@ -10,14 +10,16 @@ type Config struct {
 	PublicListener  string `mapstructure:"grpc-public-listener"`
 	PrivateServices []Service
 	PrivateListener string `mapstructure:"grpc-private-listener"`
-	TLSServices     []Service
-	TLSListener     string `mapstructure:"grpc-tls-listener"`
-	TLSCACert       string `mapstructure:"grpc-tls-ca-cert"`
-	TLSCert         string `mapstructure:"grpc-tls-cert"`
-	TLSKey          string `mapstructure:"grpc-tls-key"`
-	GrpcSendMsgSize int    `mapstructure:"grpc-send-msg-size"`
-	GrpcRecvMsgSize int    `mapstructure:"grpc-recv-msg-size"`
-	JSONListener    string `mapstructure:"grpc-json-listener"`
+	PostServices    []Service
+	PostListener    string    `mapstructure:"grpc-post-listener"`
```

Comment:
Yes, by default it only includes the `Post` service. The main reason is that setting it up in `node.go` is analogous to setting up other listeners.

Reply:
But that should probably never be re-configurable by the users and it currently is.

Reply:
No it's not 🙂, the config option is not exposed, neither via the config nor via cmdline options. To change the setting you have to recompile the node.

Reply:
It is exposed, see https://pkg.go.dev/github.com/mitchellh/mapstructure#hdr-Field_Tags:

>When decoding to a struct, mapstructure will use the field name by default to perform the mapping.
(...)
You can change the behavior of mapstructure by using struct tags.

You can set the `PostServices` config field with a `postServices` key in the  JSON config file.

Reply:
Good find! I thought it would ignore them by default, but thankfully there is an option for that:

https://pkg.go.dev/github.com/mitchellh/mapstructure#section-readme

Maybe we should set that to true? I think there are other places where config fields are not tagged with the intention of not having them configurable via config file or cmdline parameter.

Reply:
I don't think we can resolve this properly here and created https://github.com/spacemeshos/go-spacemesh/pull/5466 to discuss a general solution for it 🙂